src/core.ts

```diff
@@ -25,38 +25,42 @@ export class Core {
             await this.blockchains[blockchainType].refresh();
     }
 
-    async getPathAndRate(sourceToken: Token, targetToken: Token, amount: string = '1'): Promise<{path: Token[], rate: string}> {
+    async getPathAndRates(sourceToken: Token, targetToken: Token, amounts: string[] = ['1']): Promise<Array<{path: Token[], rate: string}>> {
         const sourceBlockchain = this.blockchains[sourceToken.blockchainType];
         const targetBlockchain = this.blockchains[targetToken.blockchainType];
 
         // single blockchain path - get the path/rate from that blockchain
         if (sourceBlockchain == targetBlockchain) {
             const paths = await this.getPaths(sourceToken.blockchainType, sourceToken, targetToken);
-            const rates = await this.getRates(sourceToken.blockchainType, paths, amount);
-            const index = Core.getBest(paths, rates);
-            return {
-                path: paths[index],
-                rate: rates[index]
-            };
+            const ratesByAmount = await this.getRates(sourceToken.blockchainType, paths, amounts);
+            return ratesByAmount.map(rates => {
+                const index = Core.getBest(paths, rates);
+                return {
```

Comment:
Why is this `return {` needed here?

Reply:
This preserves the pre-existing `return`. Each input amount corresponds to one output `{ path: Token[], rate: string}`. 

For example, if input `amounts = ['1', '10', '20']`, then output would be
```
[
    { path: best path when amount = 1, rate: rate for that path when amount = 1 },
    { path: best path when amount = 10, rate: rate for that path when amount = 10 },
    { path: best path when amount = 20, rate: rate for that path when amount = 20 },
]
```

Reply:
Oh, it's inside the callback function that you pass to `map`.
Missed that.